### PR TITLE
Feature/phonon transport new materials

### PR DESCRIPTION
**Pull request by relineha**
Requesting a merge of the branch with phonon transport in new materials (Al2O3, LiF, CaWO4, CaF2, GaAs). We have also made a small tweak to the rhombohedral crystal group implementation.

### CODE REVIEW

**Comment by kelseymh** [comment]
This pull request, and the named branch, are the implementation of JIRA ticket [G4CMP-387](https://jira.slac.stanford.edu/browse/G4CMP-387).  

* The `ChangeHistory` file should be updated using the JIRA ticket identifier "G4CMP-387".
* When this branch is merged onto the develop, the merge should be tagged "G4CMP-387".

File: library/src/G4CMPCrystalGroup.cc [hunk 1]
```diff
@@ -141,16 +140,56 @@ G4bool G4CMPCrystalGroup::FillOrthorhombic(G4double Cij[6][6]) const {
...
+  Cij[4][1] = -0.0;
+  Cij[5][3] = -0.0;
+  Cij[5][4] = C14;
+  Cij[5][5] = C66;
+  
+  Cij[3][0]=C14;
+  Cij[1][0]=C12;
+  Cij[2][0]=C13;
+  
+  ///////////Filling the other terms of the Matix with zeros (on top of those already initialized to zero and then never changed)
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:165]: Please don't use long strings of slashes, and please limit lines to 80 characters or less.

**Comment by szatschler** [comment]
The comment sections and code changes in library/include/G4CMPCrystalGroup.hh and library/src/G4CMPCrystalGroup.cc should be reviewed. In the first file, a comment has been removed in a commit (probably by accident) but the associated code appears to be still in there. In the latter, a comment line has been removed (probably again by accident) but also the associated code lines adding explicit "fallthrough" statements were removed (at least in two places). Moreover, some new comments were added to the top of the file that do not follow the usual convention.

File: examples/Phonon_Caustics/include/Caustic_PhononConfigManager.hh [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+#ifndef Caustic_PhononConfigManager_hh
+#define Caustic_PhononConfigManager_hh 1
+//20240110 Israel Hernandez -- Illinois Institute of Technology, Quantum Science Center and Fermilab
+
+#include "globals.hh"
```
Comment (kelseymh) [inline examples/Phonon_Caustics/include/Caustic_PhononConfigManager.hh:10]: Geant4 recommends avoiding the use of "globals.hh" in .hh files.  To get access to the basic data types, use
```
    #include "G4Types.hh"
    #include "G4String.hh"
```

Consider this comment as applying to all of the .hh files.

File: examples/Phonon_Caustics/g4cmpPhononCaustics.cc [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+// //
+// // //Make notes for the meetin g
+// //    UImanager->ApplyCommand("/vis/scene/add/trajectories smooth");
+// //    UImanager->ApplyCommand("/vis/scene/endOfEventAction accumulate");
+//
+//
+//     ui->SessionStart();
+
+
+
```
Comment (kelseymh) [inline examples/Phonon_Caustics/g4cmpPhononCaustics.cc:124]: Please do not leave a bunch of commented out lines in the code.

File: examples/Phonon_Caustics/g4cmpPhononCaustics.cc [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+ }
+
+ delete visManager;
+ delete runManager;
+
+ return 0;
+///////
+
+
+
```
Comment (kelseymh) [inline examples/Phonon_Caustics/g4cmpPhononCaustics.cc:150]: Please do not leave extraneous blank lines.

File: examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+{
+  G4NistManager* nistManager = G4NistManager::Instance();
+
+  fLiquidHelium = nistManager->FindOrBuildMaterial("G4_AIR"); // to be corrected
+  fBolometer = nistManager->FindOrBuildMaterial("G4_Al");
+
+//This lines works for Sapphire
+//
+ fOxigen = nistManager->FindOrBuildMaterial("G4_O");
+fSubstrate = new G4Material("fSubstrate", 3.98*g/cm3, 2);
```
Comment (kelseymh) [inline examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc:92]: Please don't use the name of your variable as the name of the material.  You're defining sapphire, so give it the name "Sapphire".  Then when some code prints out the material's `GetName()`, the output will be understandable.

File: examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+  //
+  G4VSolid* worldSolid = new G4Box("World",6.*cm,6.*cm,6.*cm);
+  G4LogicalVolume* worldLogical =
+    new G4LogicalVolume(worldSolid,fLiquidHelium,"World");
+  worldLogical->SetUserLimits(new G4UserLimits(10*mm, DBL_MAX, DBL_MAX, 0, 0));
+  fWorldPhys = new G4PVPlacement(0,G4ThreeVector(),worldLogical,"World",0,
+                                 false,0);
+
+// Substrate
+ G4VSolid* fSubstrateSolid= new G4Box("fSubstrateSolid",0.2*cm,0.2*cm,0.2*cm);
```
Comment (kelseymh) [inline examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc:114]: I would encourage you to use use human-readable and descriptive names for the volumes, rather than the names of internal variables.

File: examples/Phonon_Caustics/src/Caustic_PhononSensitivity.cc [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    //Saving in a txt file the Final Phonon Position.
+    for (G4CMPElectrodeHit* hit : *hitVec) {
+        output << hit->GetParticleName() << '\t'
+        << hit->GetFinalPosition().getX()/m << '\t'
+        << hit->GetFinalPosition().getY()/m << '\t'
+        << hit->GetFinalPosition().getZ()/m << '\n';
+
+    }
+  }
+}
```
Comment (kelseymh) [inline examples/Phonon_Caustics/src/Caustic_PhononSensitivity.cc:54]: Just as a note, this is not thread-safe.  You're going to end up either having each thread open and write to the same file, or you'll have a file opened by master and all the threads writing to it.  Either way, the result will be a mish-mash of overlapping data.  

Making this thread safe could be done straightforwardly by using G4Analaysis, defining a tuple, and having G4Analysis write out a CSV (or tab-delimited) file.

File: examples/Phonon_Caustics/src/Caustic_PhononSensitivity.cc [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+void Caustic_PhononSensitivity::SetOutputFile(const G4String &fn) {
```
Comment (kelseymh) [inline examples/Phonon_Caustics/src/Caustic_PhononSensitivity.cc:56]: See above.  This isn't thread safe.

**Review by kelseymh — CHANGES_REQUESTED** [review]
I may have missed some things.  Also, some of my comments apply everywhere, even though they're only attached in one place.

File: library/src/G4CMPCrystalGroup.cc [hunk 9]
```diff
@@ -10,6 +10,7 @@
 //
 // 20170728  Change function args "alpha, beta, gamma" to "al, bt, gm" (-Wshadow)
 // 20240426  S. Zatschler -- Add explicit fallthrough statements to switch cases
+// 20240924  I. Hernandez -- Add an if statement to include a rhombohedral Crystal group and filling more explicitly the function G4CMPCrystalGroup::FillOrthorhombic()
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:13]: Please limit comment lines to 80 columns or less (look at all the preceding entries).  Just break your comment into multiple lines.

File: library/include/G4CMPCrystalGroup.hh [hunk 10]
```diff
@@ -13,6 +13,7 @@
 // 20170525  M. Kelsey -- Add default "rule of five" copy/move operators
 // 20170728  Change function args "alpha, beta, gamma" to "al, bt, gm" (-Wshadow)
 // 20240416  S. Zatschler -- Change function arg "g" to "c" (-Wshadow)
+// 20240924  I. Hernandez -- add    G4bool FillHexagonal(G4double Cij[6][6]) const;
```
Comment (kelseymh) [inline library/include/G4CMPCrystalGroup.hh:16]: Maybe make this a bit more narrative?  "Add functionality to support hexagonal crystal structures."

File: library/src/G4CMPCrystalGroup.cc [hunk 11]
```diff
@@ -79,11 +80,9 @@ void G4CMPCrystalGroup::SetTriclinic(G4double al, G4double bt,
 G4bool G4CMPCrystalGroup::FillElReduced(G4double Cij[6][6]) const {
   switch (group) {
   case amorphous:    Cij[3][3] = 0.5*(Cij[0][0]-Cij[0][1]); // Cubic, C44 set
-                     [[fallthrough]];
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:82]: Why did you remove all of the `[[fallthrough]]` directives?  Those are needed to avoid a whole lot of compiler warnings.

File: library/src/G4CMPCrystalGroup.cc [hunk 12]
```diff
@@ -141,15 +140,48 @@ G4bool G4CMPCrystalGroup::FillOrthorhombic(G4double Cij[6][6]) const {
...
   G4double C11=Cij[0][0], C12=Cij[0][1], C13=Cij[0][2], C14=Cij[0][3];
-  G4double C15=Cij[0][4], C33=Cij[2][2], C44=Cij[3][3], C66=0.5*(C11-C12);
-
-  Cij[1][1] = C11;	// Copy small number of individual elements
-  Cij[1][2] = C13;
-  Cij[1][3] = -C14;
-  Cij[1][4] = -C15;
-  Cij[3][5] = -C15;
-  Cij[4][4] = C44;
-  Cij[4][5] = C14;
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:152]: This looks like a substantial bug fix?  Thank you!  In your dated comment at the top, could you please provide a _narrative_ explanation of this, rather than just naming the function.

**Review by kelseymh — CHANGES_REQUESTED** [review]

File: library/src/G4CMPCrystalGroup.cc (hunk 11 above)
Comment (kelseymh) [reply library/src/G4CMPCrystalGroup.cc:82]: In particular, these are the code changes which Stefan referenced in the comment dated 20240426.  Did you restore the comment "by hand", without restoring the missing code?

File: examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc [hunk 13]
```diff
@@ -145,8 +145,8 @@ G4VSolid* SubstrateSolid= new G4Box("SubstrateSolid",0.2*cm,0.2*cm,0.2*cm);
 
 G4LogicalVolume* BolometerLogical =
```
Comment (kelseymh) [inline examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc:146]: You don't need to change it, but why exactly did you remove the leading "f" from the name of the variable?  That is a standard convention throughout the Geant4 software, and I was not suggesting that you change it.  I was _only_ suggesting that you change the *string* name assigned to the various volumes.  Those *strings* (not the local C++ variable names) can be printed out using standard Geant4 macro commands:
```
/vis/open ATree
/vis/ASCIITree/verbose 12
/vis/drawTree worlds
```
so the *strings* are useful as human readable documentation for the geometry that is being built.

Don't bother changing things back.  This has already dragged on too long.  I'm sorry that I wasn't completely clear about what I was asking for.

File: library/src/G4CMPCrystalGroup.cc [hunk 14]
```diff
@@ -10,7 +10,8 @@
 //
 // 20170728  Change function args "alpha, beta, gamma" to "al, bt, gm" (-Wshadow)
 // 20240426  S. Zatschler -- Add explicit fallthrough statements to switch cases
-// 20240924  I. Hernandez -- Add an if statement to include a rhombohedral Crystal group and filling more explicitly the function G4CMPCrystalGroup::FillOrthorhombic()
+// 20241014  I. Hernandez -- Add an if statement to include a rhombohedral Crystal group.
+//           Filling more explicitly the function G4CMPCrystalGroup::FillOrthorhombic()
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:14]: This line is still longer than 80 characters.  Does your editor support specifying a column after which to automatically insert a line break?

File: library/src/G4CMPCrystalGroup.cc [hunk 15]
```diff
@@ -10,7 +10,8 @@
 //
 // 20170728  Change function args "alpha, beta, gamma" to "al, bt, gm" (-Wshadow)
 // 20240426  S. Zatschler -- Add explicit fallthrough statements to switch cases
-// 20240924  I. Hernandez -- Add an if statement to include a rhombohedral Crystal group and filling more explicitly the function G4CMPCrystalGroup::FillOrthorhombic()
+// 20241014  I. Hernandez -- Add an if statement to include a rhombohedral Crystal group.
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:13]: This line appears to have a whole lot of blanks (space characters, ASCII 0x20) following the period.  Please remove all of them.

File: library/src/G4CMPCrystalGroup.cc [hunk 16]
```diff
@@ -141,7 +144,9 @@ G4bool G4CMPCrystalGroup::FillOrthorhombic(G4double Cij[6][6]) const {
 G4bool G4CMPCrystalGroup::FillRhombohedral(G4double Cij[6][6]) const {
   G4double C11=Cij[0][0], C12=Cij[0][1], C13=Cij[0][2], C14=Cij[0][3];
   G4double  C33=Cij[2][2], C44=Cij[3][3], C66=0.5*(C11-C12);
-//Filling the other terms of the matrix, one-by-one (IH)
+//Filling the second order elastic constant for Rhombohedral I Class with Point group 32, 3m and 3m.
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:147]: This line is longer than 80 characters.  Please insert a break after the word "Point", so that a new line starts with "// group".  Please insert a space before the uppercase "F" at the beginning of the comment.

File: library/src/G4CMPCrystalGroup.cc [hunk 17]
```diff
@@ -141,7 +144,9 @@ G4bool G4CMPCrystalGroup::FillOrthorhombic(G4double Cij[6][6]) const {
...
+//Filling explicitly the  36 components of the second order elastic tensor.
```
Comment (kelseymh) [inline library/src/G4CMPCrystalGroup.cc:148]: Please insert a space before the uppercase "F" at the beginning of the comment.

**Review by kelseymh — CHANGES_REQUESTED** [review]
Mostly requests to format comments in a readable way.

File: examples/Phonon_Caustics/README.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+```console
+User@fun:~$ cmake ..
+User@fun:~$ make install
+[100%] Built target g4cmpPhononCaustics
+```
+If you do not have errors
+```console
+./g4cmpaSapphire Caustic.mac
+```
+You will have a txt file "phonon_hits.txt" with three columns: <br>
```
Comment (kelseymh) [inline examples/Phonon_Caustics/README.md:54]: Please include a comment, either here or somewhere nearby, that the example does not currently support running with multiple threads.  The user should ensure that `/run/numberOfThreads 1` is specified in their macro file.

File: examples/Phonon_Caustics/Caustic.mac [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+/run/printProgress 100000
+/process/inactivate phononScattering
+/process/inactivate phononDownconversion
+/g4cmp/phononBounces 3
+#/tracking/verbose 2
+/gps/energy 0.03 eV
+/gps/number 1
+/gps/ang/type iso
+/gps/pos/centre 0.00 0.0 0.0 cm
+/run/beamOn 40000000
```
Comment (kelseymh) [inline examples/Phonon_Caustics/Caustic.mac:13]: Please add the required `/run/numberOfThreads 1` here, before `beamOn`, because the text-based output is not thread-safe.

File: examples/Phonon_Caustics/g4cmpPhononCaustics.cc [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+/// \file phonon/g4cmpPhonon.cc
+/// \brief Main program of the G4CMP/phonon example
+//
+// $Id$
+//
+// 20140509  Add conditional code for Geant4 10.0 vs. earlier
+// 20150112  Remove RM->Initialize() call to allow macro configuration
+// 20160111  Remove Geant4 version check since we now hard depend on 10.2+
+// 20170816  Add example-specific configuration manager
+// 20220718  Remove obsolete pre-processor macros G4VIS_USE and G4UI_USE
```
Comment (kelseymh) [inline examples/Phonon_Caustics/g4cmpPhononCaustics.cc:15]: Since this is a new and different file, please remove the past history that you copied from elsewhere.

File: examples/Phonon_Caustics/g4cmpPhononCaustics.cc [hunk 21]
```diff
@@ -0,0 +1,151 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+/// \file phonon/g4cmpPhonon.cc
+/// \brief Main program of the G4CMP/phonon example
```
Comment (kelseymh) [inline examples/Phonon_Caustics/g4cmpPhononCaustics.cc]: Update these lines to reflect the actual file and directory.

**Review by kelseymh — CHANGES_REQUESTED** [review]

File: examples/Phonon_Caustics/Caustic.mac [hunk 22]
```diff
@@ -0,0 +1,13 @@
...
+/run/initialize
+/run/printProgress 100000
+/process/inactivate phononScattering
+/process/inactivate phononDownconversion
+/g4cmp/phononBounces 3
+#/tracking/verbose 2
+/gps/energy 0.03 eV
+/gps/number 1
+/gps/ang/type iso
+/gps/pos/centre 0.00 0.0 0.0 cm
```
Comment (kelseymh) [inline examples/Phonon_Caustics/Caustic.mac:12]: I would suggest moving these lines up to the beginning of your macro, before processes and worker threads get instantiated.  That way all of the final "after /run/init" stuff is at the end of the macro, right before the run starts.

**Review by kelseymh — CHANGES_REQUESTED** [review]

File: examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc (hunk 13 above)
Comment (Israel-Tanjiro) [reply examples/Phonon_Caustics/src/Caustic_PhononDetectorConstruction.cc:146]: I changed the variable's name because it is not a global variable defined in the Class.

File: library/src/G4CMPCrystalGroup.cc (hunk 14 above)
Comment (Israel-Tanjiro) [reply library/src/G4CMPCrystalGroup.cc]: I added the following line. I decided to break it into two comments because this is how I solved this bug. 
First,  I added the if statement, and later I fixed the function  FillOrthorhombic()
// 20241024  I. Hernandez -- Add an if statement to include a rhombohedral Crystal group
// 20241024  I. Hernandez -- Fixing the function FillOrthorhombic()

File: library/src/G4CMPCrystalGroup.cc (hunk 16 above)
Comment (Israel-Tanjiro) [reply library/src/G4CMPCrystalGroup.cc]: I change those lines like this :
// Filling the second order elastic constant for Rhombohedral I Class with
// Point group 32, 3m and 3m.
// Filling explicitly the  36 components of the second order elastic tensor.

File: examples/Phonon_Caustics/README.md (hunk 18 above)
Comment (Israel-Tanjiro) [reply examples/Phonon_Caustics/README.md]: I include the following statement on the Readme file 
This example does not support running with multiple threads. The user should ensure that  /run/numberOfThreads 1  is specified in their macro file.

File: examples/Phonon_Caustics/Caustic.mac (hunk 22 above)
Comment (Israel-Tanjiro) [reply examples/Phonon_Caustics/Caustic.mac]: I changed the macro to this:
# This Macro is to Generate Phonons Caustics patterns
/gps/energy 0.03 eV
/gps/number 1
/gps/ang/type iso
/gps/pos/centre 0.0 0.0 0.0 cm
/run/numberOfThreads 1
/run/initialize
/g4cmp/phononBounces 100
/process/inactivate phononScattering
/process/inactivate phononDownconversion
/run/beamOn 40000
 
This is the only way that the program works

File: examples/Phonon_Caustics/g4cmpPhononCaustics.cc (hunk 20 above)
Comment (Israel-Tanjiro) [reply examples/Phonon_Caustics/g4cmpPhononCaustics.cc]: I removed all the history and now the program starts with the following lines:
/***********************************************************************\
 * This software is licensed under the terms of the GNU General Public *
 * License version 3 or later. See G4CMP/LICENSE for the full license. *
\***********************************************************************/

/// \file Phonon_Caustics/g4cmpPhononCaustics.cc
/// \brief Main program of the G4CMP/Phonon_Caustics example
//
// $Id$
//

File: examples/Phonon_Caustics/Caustic.mac (hunk 22 above)
Comment (kelseymh) [reply examples/Phonon_Caustics/Caustic.mac]: That's odd!  In the CDMS simulations, the `/g4cmp/...` commands can be anywhere; we instantiate {{G4CMPConfigManager}} in our main(), just like you do (I thought! Let me check...)  Yes, you have that at [line 42](https://github.com/kelseymh/G4CMP/blob/82b5254362920699c03675eab17c9f5c42609e2d/examples/Phonon_Caustics/g4cmpPhononCaustics.cc#L42).  Is it really true that the `phononBounces` command doesn't work before `/run/initialize`?

Comment (Israel-Tanjiro) [reply examples/Phonon_Caustics/Caustic.mac]: You are right. For this example, this command is not necessary, because the phonons are detected at the first hit. I have eliminated this command.

**Review by kelseymh — APPROVED** [review]
